processing/src/main/java/org/apache/druid/segment/join/table/IndexedTableJoinMatcher.java

```diff
@@ -262,8 +288,9 @@ public ValueType defaultType()
         final IndexedInts row = selector.getRow();
 
         if (row.size() == 1) {
-          final String key = selector.lookupName(row.get(0));
-          return index.find(key).iterator();
+          int dimensionId = row.get(0);
+          //noinspection ConstantConditions (cache cannot return nulls since nulls are never stored in cache)
+          return dimensionCaches.get(selector).get(dimensionId).iterator();
```

Comment:
This is only safe when the selector has a 'real' dictionary, i.e. when `selector.getValueCardinality()` does not return `DimensionSelector.CARDINALITY_UNKNOWN`. If it is unknown then the ids are not valid outside the context of a specific row. So, in that case you'll need to fall back to the slower code.